curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb2e5bdfb23a67272a07c7.md

```diff
@@ -0,0 +1,391 @@
+---
+id: 64cb2e5bdfb23a67272a07c7
+title: Step 93
+challengeType: 0
+dashedName: step-93
+---
+
+# --description--
+
+Now you need to create a `draw` method for the `CheckPoint` class. 
+
+Inside the `draw` method, add a `fillStyle` property to the `ctx` object and set it to `#f1be32`. 
+
+Below the `fillStyle` property, use the `fillRect` method on the `ctx` object and pass in the `x`, `y`, `width`, and `height` properties as parameters.
+
+# --hints--
+
+Your `CheckPoint` class should have a `draw` method.
+
+```js
+const splitter = code.split("ctx.fillRect(this.position.x, this.position.y, this.width, this.height)")
+assert.match(splitter[2], /draw\(\)\s*\{/);
+```
+
+Your `draw` method should have a `fillStyle` property.
+
+```js
+const splitter = code.split("ctx.fillRect(this.position.x, this.position.y, this.width, this.height)")
+assert.match(splitter[2], /draw\(\)\s*\{\s*ctx\.fillStyle\s*=\s*('|")#f1be32\1;?/);
+```
+
+Your `draw` method should have a `fillRect` method.
+
+```js
+const splitter = code.split('ctx.fillStyle = "#f1be32";')
```

Comment:
This will give troubles when somebody uses single quotes, wouldn't splitting only on `#f1be32` work?